### PR TITLE
Add "copy (absolute) path" menu items for tabs

### LINKED ISSUE #13970: "Copy (absolute) path" context menu option for tabs
### Check for existing issues

- [X] Completed

### Describe the feature

Related to https://github.com/zed-industries/zed/pull/11181

It would be useful to be able to easily copy the relative and absolute path of any file open in a tab.

From my initial inspection it seems somewhat difficult in the current, fairly abstract model of "project items", but it may be possible, such as by adding a new method to the Item trait? I might need some guidance here.

### If applicable, add mockups / screenshots to help present your vision of the feature

See similar function in VS Code

### PR DESCRIPTION
**Pull request by mattfbacon**
So far we have the basic functionality, but it doesn't work on all tabs because some of them don't properly implement for_each_project_item, most notably the image viewer.

Closes #13970. Related to #11181.

Release Notes:

- Added context menu items for tabs to copy the file's path ([#13970](https://github.com/zed-industries/zed/issues/13970)).

An example of the updated context menu showing "Copy path" and "Copy absolute path"

### CODE REVIEW

**Comment by mattfbacon** [comment]
For the image viewer, I think it would be better to work just based on the project path, but I don't know how to proceed with modifying the persistence DB and that kind of thing.

File: crates/workspace/src/pane.rs [hunk 1]
```diff
@@ -1786,14 +1795,36 @@ impl Pane {
                                     });
                                 }),
                             )
-                            .when_some(parent_abs_path, |menu, abs_path| {
+                            .when_some(rel_path, |menu, path| {
+                                menu.entry(
+                                    "Copy Path",
```
Comment (williamdes) [inline crates/workspace/src/pane.rs:1800]:
```suggestion
                                    "Copy Relative Path",
```
Like vscode IIRC ?

Comment (mattfbacon) [reply crates/workspace/src/pane.rs:1800]: I considered that, but I'm not sure it's always correct if a file was opened with an absolute path to begin with. What does vs code do in that case?

Comment (williamdes) [reply crates/workspace/src/pane.rs:1800]: > I considered that, but I'm not sure it's always correct if a file was opened with an absolute path to begin with. What does vs code do in that case?

That's true, what can we do if the user opens files out of the workspace.
IIRC it is possible in VScode but not in Zed for now.

Comment (mattfbacon) [reply crates/workspace/src/pane.rs:1800]: Ah good point, nevermind then.

File: crates/workspace/src/pane.rs [hunk 2]
```diff
@@ -5,17 +5,17 @@ use crate::{
...
 use gpui::{
     actions, anchored, deferred, impl_actions, prelude::*, Action, AnchorCorner, AnyElement,
-    AppContext, AsyncWindowContext, ClickEvent, DismissEvent, Div, DragMoveEvent, EntityId,
-    EventEmitter, ExternalPaths, FocusHandle, FocusOutEvent, FocusableView, KeyContext, Model,
-    MouseButton, MouseDownEvent, NavigationDirection, Pixels, Point, PromptLevel, Render,
+    AppContext, AsyncWindowContext, ClickEvent, ClipboardItem, DismissEvent, Div, DragMoveEvent,
+    EntityId, EventEmitter, ExternalPaths, FocusHandle, FocusOutEvent, FocusableView, KeyContext,
+    Model, MouseButton, MouseDownEvent, NavigationDirection, Pixels, Point, PromptLevel, Render,
     ScrollHandle, Subscription, Task, View, ViewContext, VisualContext, WeakFocusHandle, WeakView,
     WindowContext,
```
Comment (williamdes) [inline crates/workspace/src/pane.rs:20]: it wants a run about formatting

**Comment by SomeoneToIgnore** [comment]
Thank you, this PR looks quite stale and https://github.com/zed-industries/zed/pull/15260 comes as a new attempt, so I'll close this one.

**Pull request closed without merging**